src/Item_Kanban.php

```diff
@@ -54,14 +54,19 @@ public static function saveStateForItem($itemtype, $items_id, $state)
         /** @var \DBmysql $DB */
         global $DB;
 
-        /** @var Kanban|CommonDBTM $item */
```

Comment:
I do no know this part of the code; @cconard96 could you please review this change?

Reply:
Kanban is a trait, there doesn't seem to be any syntax to indicate it correctly.

Reply:
My remark concerns the whole change, not just phpdoc

Reply:
`Kanban&CommonDBTM` is the more correct way as far as I know.
Not sure why this change was made.

Should just change the type in the PHPDoc for `$itemtype` to `class-string<Kanban&CommonDBTM>`. No need for the method checks.

Reply:
Well, maybe not... I may have been right the first time.

https://phpstan.org/blog/union-types-vs-intersection-types

Reply:
Traits are not allowed to be type checked (you can't use stuff like instance of on classes that uses them), this is probably why PHPStan does not accept it as a valid type.

Reply:
The correct way to deal with this is to have an interface, which is satisfied by the trait.
You can then use the interface type (and check it with instanceOf).

Reply:
So again, traits are a terrible language feature and just about every tool has garbage support for them, if any.

At least with the way it was before, PHPStorm recognized that the items were CommonDBTM and had the Kanban trait so it knew what methods were available.

Reply:
This proposed change is OK for me, as long as the `method_exist()` check is made on the trait methods existence.